index.js

```diff
@@ -0,0 +1,68 @@
+const addon = require('bindings')('module');
+
+const pkcs7_header = Buffer.from('-----BEGIN PKCS7-----\n');
+const pkcs7_footer = Buffer.from('\n-----END PKCS7-----\n');
+const nl = '\n'.charAt(0);
+
+/**
+ * Verify a document given a public key, document and a PKCS#7 encoded
+ * signature.  All parameters should either be strings, Buffers or something
+ * which can be converted into a Buffer safely with a call to Buffer.from(),
+ * with an encoding parameter of 'utf-8'.
+ */
+module.exports = function verify(pubkey, document, pkcs7) {
+  if (typeof pubkey === 'undefined') {
+    throw new Error('pubkey must be provided');
+  }
+
+  if (typeof document === 'undefined') {
+    throw new Error('document must be provided');
+  }
+
+  if (typeof pkcs7 === 'undefined') {
+    throw new Error('pkcs7 must be provided');
+  }
+
+  if (!Buffer.isBuffer(pubkey)) {
+    pubkey = Buffer.from(pubkey, 'utf-8');
+  }
+
+  if (!Buffer.isBuffer(document)) {
+    document = Buffer.from(document, 'utf-8');
+  }
+
+  if (!Buffer.isBuffer(pkcs7)) {
+    pkcs7 = Buffer.from(pkcs7, 'utf-8');
+  }
+
+  hasHeader = true;
+  for (let i = 0; i < pkcs7_header.length && hasHeader; i++) {
+    if (pkcs7[i] !== pkcs7_header[i]) {
+      hasHeader = false;
+    }
+  }
```

Comment:
I'll change to using Buffer.compare.  The compare method on the instance was working in a strange way, but I doubt one is faster than the other, since .slice() should be a close-to-zero operation since it shares the same backing memory.